### DISCUSSION TITLE
Path undefined error when using multiple flat layouts

### DISCUSSION THREAD

**Discussion by seanmtaylor** [remix-run/remix#1206] [Q&A] [answered]
I'm trying to setup some flat layouts so that I can have layouts for the marketing section and app section of my site, just like the example in the docs - https://remix.run/docs/en/v1/api/conventions#flat-layout-routes

I can create a single flat layout file just fine.

But as soon as I create the second layout file, I get the following error:

```
Path undefined defined by route "routes/__app" conflicts with route "routes/__marketing"
```

I've created a fresh remix app to make sure it wasn't something funky I was doing, but the problem still persists.

My file structure is as follows (I've ommited the default remix files as I haven't touched them):

```
app/
├── routes/
│   ├── __app/
│   │   ├── dashboard.tsx
│   └── __marketing
│   │   ├── index.tsx
│   ├── __app.tsx
│   ├── __marketing.tsx
```

The contents of the files are:

```jsx
// routes/__app/dashboard.tsx
export default function Dashboard() {
    return (<h1>Dashboard</h1>);
}
```
```jsx
// routes/__app.tsx
import { Outlet } from "remix";

export default function AppLayout() {
    return (<Outlet />);
}
```

```jsx
// routes/__marketing/index.tsx
export default function Index() {
  return (<h1>Index</h1>);
}

```
```jsx
// routes/__marketing.tsx
import { Outlet } from "remix";

export default function MarketingLayout() {
    return (<><Outlet /></>);
}
```

I've kept them as basic as possible.

What am I doing wrong? Any help would be greatly appreciated.

**Comment by hollandThomas** [comment]
Can confirm the behavior with latest Node 14, 16 and 17

This seems to be a bug to me, probably somewhere in this file (maybe the comment is right and there should be some tests added^^)
https://github.com/remix-run/remix/blob/1fd70960e4d88740df5bf407a6ba2cd2b9549459/packages/remix-dev/config/routesConvention.ts#L115

**Reply by seanmtaylor** [reply]
Thanks for confirming that it's an issue @hollandThomas

**Comment by seanmtaylor** [comment, marked as answer]
Looks like an [issue](https://github.com/remix-run/remix/issues/1210) and [potential fix](https://github.com/remix-run/remix/pull/1216) have been put in 🙏

**Reply by ryanflorence** [reply]
Yep, this is a bug
